src/syngen/ml/train_chain/train_chain.py

```diff
@@ -236,17 +238,22 @@ def run(self, size: int, run_parallel: bool = True):
     def kde_gen(self, pk_table, pk_column_label, size, fk_label):
         pk = pk_table[pk_column_label]
 
-        if pk.dtype == "object":
-            synth_fk = pk.sample(size, replace=True).reset_index(drop=True)
-            synth_fk.rename(fk_label, inplace=True)
-        else:
-            with open(self.fk_kde_path, "rb") as file:
+        try:
+            with open(f"{self.fk_kde_path}_{fk_label}.pkl", "rb") as file:
                 kde = dill.load(file)
             pk = pk.dropna()
-            fk_pdf = kde.evaluate(pk)
+            if pk.dtype == "object":
```

Comment:
Very minor comment: you can do something like
numeric_pk = np.arange(len(pk)) if pk.dtype == "object" else pk
fk_pdf = np.maximum(kde.evaluate(numeric_pk ), 1e-12)